queries/smali/highlights.scm

```diff
@@ -0,0 +1,112 @@
+; Types
+
+(class_identifier) @type
+
+(primitive_type) @type.builtin
+
+(array_type
+  "[" @punctuation.bracket)
+
+; Methods
+
+(method_declaration
+  (method_identifier) @method)
+
+(statement
+  (opcode) @_invoke
+  (full_method_identifier
+    (method_identifier) @method.call)
+  (#lua-match? @_invoke "^invoke"))
+
+(method_identifier
+  "<init>(" @constructor
+  ")" @constructor)
+
+(method_identifier
+  "<clinit>(" @constructor
+  ")" @constructor)
+
+; Fields
+
+(field_identifier) @field
+
+; Parameters
+
+(parameter) @parameter
+
+; Variables
+
+(variable) @variable
+
+; Labels
+
+(label) @label
+
+; Operators
+
+(opcode) @keyword.operator
+
+((opcode) @keyword.return
+ (#lua-match? @keyword.return "^return"))
+
+"=" @operator
+
+; Keywords
+
+[
+  ".class"
+  ".super"
+  ".source"
+  ".implements"
+  ".field"
+  ".method"
+  ".annotation"
+  ".subannotation"
+  ".param"
+  ".line"
+  ".locals"
+  ".registers"
+  ".catch"
+  ".catchall"
+  ".packed-switch"
+  ".end packed-switch"
+  ".sparse-switch"
+  ".end sparse-switch"
+  ".array-data"
+  ".end array-data"
+  ".enum"
+] @keyword
+
+[
+  (end_field)
+  (end_method)
```

Comment:
I don't know Smali. Could you check whether `@keyword.function` apply to any of these (.e.g `end_method`)

Reply:
It could be used for `.method` and `end_method` but I think it'd be better to keep them consistent with the rest.

Reply:
.method and .end_method make sense to me to have as @keyword.function, even if syntactically it's the same as the others

Reply:
Up to you.